File: ytrue-infra/ytrue-infra-core/src/main/java/com/ytrue/infra/core/base/BaseCodeException.java
package com.ytrue.infra.core.base;

import com.ytrue.infra.core.response.ServerResponseInfoEnum;
import lombok.Getter;

import java.io.Serial;

@Getter
public abstract class BaseCodeException extends RuntimeException {

    @Serial
    private static final long serialVersionUID = 5873796476107436739L;
    /**
     * 状态码
     */
    private String code = ServerResponseInfoEnum.INTERNAL_SERVER_ERROR.code();

    public BaseCodeException() {
    }

    public BaseCodeException(String message) {
        super(message);
    }

    public BaseCodeException(String code, String message) {
        super(message);
        this.code = code;
    }

    public BaseCodeException(String message, Throwable cause) {
        super(message, cause);
    }

    public BaseCodeException(IServerResponseInfo responseCode) {
        super(responseCode.message());
        this.code = responseCode.code();
    }

}

File: ytrue-admin-server-csharp/Ytrue.Infrastructures/Bases/BaseCodeException.cs
using Ytrue.Infrastructures.Response;

namespace Ytrue.Infrastructures.Bases;

public abstract class BaseCodeException : Exception
{
    /// <summary>
    /// 状态吗
    /// </summary>
    public string Code { set; get; } = "500";


    protected BaseCodeException()
    {
    }

    protected BaseCodeException(string? message) : base(message)
    {
    }


    protected BaseCodeException(string code, string message) : base(message)
    {
        Code = code;
    }

    protected BaseCodeException(string? message, Exception? innerException) : base(message, innerException)
    {
    }


    protected BaseCodeException(IServerResponseInfo responseCode) : base(responseCode.Message())
    {
        Code = responseCode.Code();
    }
}